src/Lykke.Service.LykkeAdapter.Core/Handlers/OrderBookHandlerDecorator.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Lykke.Service.LykkeAdapter.Core.Domain.OrderBooks;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Lykke.Service.LykkeAdapter.Core.Handlers
+{
+    public class OrderBookHandlerDecorator : IHandler<LykkeOrderBook>
+    {
+        private const string Name = "lykke";
+        private readonly IHandler<TradingOrderBook> _rabbitMqHandler;
+        private readonly Dictionary<string, LykkeOrderBook> _halfOrderBooks = new Dictionary<string, LykkeOrderBook>();
```

Comment:
I don't understand what is the half of the orderbook.

Reply:
This is because orderbooks from Lykke come for buys and sells separately. For example, if the adapter receives orderbook where IsBuy == false for a particular instrument, it will keep it until it receives the other "half" - IsBuy == false,  when it joins them together and then publishes the orderbook then to rabbit. I agree its a bit  strange code - I left as it as, its inherited from the original implementation from ExchangeConnector.

Reply:
If it just "as-is", I'm not sure that it needs to be reviewed :-)

Reply:
So I think it's up to you - rewrite that code or not.

Reply:
this particular part is "as-is". I will see if there is a better way of handling "half" orderbooks and consider re-writing or not. The only change I introduced for Lykke adapter is the grouping of orders of the same price and suming up their volumes.